client/src/Pos/Client/Txp/Util.hs

```diff
@@ -304,55 +304,50 @@ type InputPickingWay = Utxo -> TxOutputs -> Coin -> Either TxError FlatUtxo
 -- Simple inputs picking
 -------------------------------------------------------------------------
 
-data InputPickerState = InputPickerState
-    { _ipsMoneyLeft        :: !Coin
-    , _ipsAvailableOutputs :: !FlatUtxo
-    }
-
-makeLenses ''InputPickerState
-
-type InputPicker = StateT InputPickerState (Either TxError)
-
 plainInputPicker :: PendingAddresses -> InputPickingWay
 plainInputPicker (PendingAddresses pendingAddrs) utxo _outputs moneyToSpent =
-    evalStateT (pickInputs []) (InputPickerState moneyToSpent sortedUnspent)
+    case foldl' pick ([], moneyToSpent) sortedUnspent of
+        (inps, moneyLeft)
+            | moneyLeft == mkCoin 0 ->
+                pure inps
+            | otherwise ->
+                throwError (NotEnoughMoney moneyLeft)
```

Comment:
I refactored this implementation to be simpler. No need for `StateT s (Except e)` when `foldl'` does what you need. I can revert this change if we want; it was mostly an exercise in understanding the code :)

Reply:
I'd be great to have this comment in the commit message :p Would help to understand why this commit during a merge conflict or a refactoring!

Reply:
My two cents on commits messages: A first line which is concise and give an overview of what the commit does and if needed, one blank line, and as many lines of comments as you want that describe things you can't solely express with code or in-code comment (intention, choices and reasoning). 
It's actually really helpful when digging into old code that is somewhat hard to get. Sometimes, a little note from the author help to understand a change.

Reply:
Yup, I also vote for the commit message scheme outlined by Matthias here.

Writing long and descriptive commit messages is certainly tedious, but it pays out immensely when trying to reason about code later on :)

Reply:
I like this as well! I've been getting in the habit of "Squash and merge" with my PRs and I'll annotate the commit messages more thoroughly there. Unless you'd rather I not squash+merge :sweat_smile:

Reply:
> Unless you'd rather I not squash+merge

There are good reasons to do a normal merge vs a squash+merge. I tend to do those on a case-by-case analysis preferring the normal one (thus the need for a clean history) merely because it's what `git-flow` seems to dictate 😀